GDJS/Runtime/spriteruntimeobject.js

```diff
@@ -13,31 +69,51 @@
  * @memberof gdjs
  * @class SpriteAnimationFrame
  * @param {gdjs.ImageManager} imageManager The game image manager
- * @param {Object} frameData the data to be used to create the frame.
+ * @param {FrameData} frameData The frame data used to initialize the frame
  */
+
 gdjs.SpriteAnimationFrame = function(imageManager, frameData)
 {
+    /** @type {string} */
     this.image = frameData ? frameData.image : ""; //TODO: Rename in imageName, and do not store it in the object?
+    /** @type {PIXI.Texture} */
     this.texture = gdjs.SpriteRuntimeObjectRenderer.getAnimationFrame(imageManager, this.image);
 
-    if ( this.center === undefined ) this.center = { x:0, y:0 };
-    if ( this.origin === undefined ) this.origin = { x:0, y:0 };
+    if ( this.center === undefined ) {
+        /** @type {Point} */
+        this.center = { x:0, y:0 };
+    };
+    if ( this.origin === undefined ) {
+        /** @type {Point} */
+        this.origin = { x:0, y:0 }
+    };
+    /** @type {boolean} */
     this.hasCustomHitBoxes = false;
-    if ( this.customHitBoxes === undefined ) this.customHitBoxes = [];
-    if ( this.points === undefined ) this.points = new Hashtable();
+    if ( this.customHitBoxes === undefined ) {
+        /** @type {Array<gdjs.Polygon>} */
+        this.customHitBoxes = []
+    };
+    if ( this.points === undefined ){
+        /** @type {Hashtable} */
+        this.points = new Hashtable();
+    };
     else this.points.clear();
 
     //Initialize points:
 	for(var i = 0, len = frameData.points.length;i<len;++i) {
+        /** @type {CustomPointData} */
 		var ptData = frameData.points[i];
 
+        /** @type {Point} */
         var point = {x:parseFloat(ptData.x), y:parseFloat(ptData.y)};
         this.points.put(ptData.name, point);
     }
+    /** @type {Point} */
```

Comment:
It seems redundant to type the temporary variables if their type can be inferred from originPoint?

Reply:
I think it can help whoever wants to contribute directly